sys/include/net/gnrc/netif.h

```diff
@@ -56,12 +57,21 @@ extern "C" {
  */
 typedef struct gnrc_netif_ops gnrc_netif_ops_t;
 
+/**
+ * @brief   Netif bootstrap function prototype for netdev stack
+ *
+ * Called as a starting point for the netif threads. Expected to call the
+ * thread event handler @ref gnrc_netif_thread.
+ */
+typedef thread_task_func_t gnrc_netif_bootstrap_t;
+
 /**
  * @brief   Representation of a network interface
  */
 typedef struct {
     const gnrc_netif_ops_t *ops;            /**< Operations of the network interface */
     netdev_t *dev;                          /**< Network device of the network interface */
+    netdev_t *hwdev;                        /**< Hardware layer of the network device */
```

Comment:
Either change name of `hwdev` or the name and doc on `dev` (that is the top-layer of netdev). `hwdev` can be misleading if you have "virtual" devices.